### PR TITLE
Added some initial util classes

### PR DESCRIPTION
**Pull request by f2rf2r**

### CODE REVIEW

File: commonutils/src/main/java/io/intrepid/commonutils/IntrepidJavaUtils.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+public class IntrepidJavaUtils {
+    private final static char[] hexArray = "0123456789abcdef".toCharArray();
+
+    @NonNull
+    public static String bytesToHexString(@NonNull byte[] bytes) {
+        char[] hexChars = new char[bytes.length * 2];
+        for (int j = 0; j < bytes.length; j++) {
+            int v = bytes[j] & 0xFF;
+            hexChars[j * 2] = hexArray[v >>> 4];
+            hexChars[j * 2 + 1] = hexArray[v & 0x0F];
```
Comment (ryanmeador) [inline commonutils/src/main/java/io/intrepid/commonutils/IntrepidJavaUtils.java]: Does this have a significant performance advantage over String.format()?  If performance is the reason for it being so opaque, I think there are further improvements we can make.

Comment (f2rf2r) [inline commonutils/src/main/java/io/intrepid/commonutils/IntrepidJavaUtils.java]: It wasn't for the performance reasons, I just haven't thought about using String.format(). I'm fine with going with that approach if you think it's more readable.

File: commonutils/src/main/java/io/intrepid/commonutils/IntrepidAndroidUtils.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+     * Converts dp unit to equivalent pixels, depending on device density.
+     *
+     * @param context Context to get resources and device specific display metrics
+     * @param dp      A value in dp (density independent pixels) unit. Which we need to convert into pixels
+     * @return A float value to represent px equivalent to dp depending on device density
+     */
+    public static float convertDpToPixel(@NonNull Context context, float dp) {
+        Resources resources = context.getResources();
+        DisplayMetrics metrics = resources.getDisplayMetrics();
+        return dp * (metrics.densityDpi / 160f);
```
Comment (russhwolf) [inline commonutils/src/main/java/io/intrepid/commonutils/IntrepidAndroidUtils.java]: There are APIs in `TypedValue` to do this sort of thing.
`TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, metrics);`

File: commonutils/publish.gradle [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+ext.localProperties = new Properties()
+ext.localProperties.load(project.rootProject.file('local.properties').newDataInputStream())
+
+ext.getLocalProperty = { propertyName ->
+    def property = localProperties.getProperty(propertyName)
+    return property == null ? "" : property
+}
+
+ext.artifactPath = "build/outputs/aar/commonutils-release.aar"
```
Comment (streetsofboston) [inline commonutils/publish.gradle]: Is this 'hard-coded' path robust enough? Could there be an already existing variable/field holding (part of) this path.

File: commonutils/src/main/java/io/intrepid/commonutils/IntrepidJavaUtils.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.List;
+
+public class IntrepidJavaUtils {
+
+    @NonNull
+    public static String bytesToHexString(@NonNull byte[] bytes) {
```
Comment (streetsofboston) [inline commonutils/src/main/java/io/intrepid/commonutils/IntrepidJavaUtils.java:17]: Is this big or little endian?
If big-endian, consider using BigInteger instead: 
https://developer.android.com/reference/java/math/BigInteger.html#BigInteger(int, byte[])
and then calling
https://developer.android.com/reference/java/math/BigInteger.html#toString(int)

Comment (f2rf2r) [inline commonutils/src/main/java/io/intrepid/commonutils/IntrepidJavaUtils.java:17]: I tried using the BigInteger version, but its result doesn't include the leading 0.

Comment (streetsofboston) [inline commonutils/src/main/java/io/intrepid/commonutils/IntrepidJavaUtils.java:17]: I see.
Just be sure to add some java-doc stating the byte-order of the `bytes` array.

File: commonutils/publish.gradle (hunk 3 above)
Comment (f2rf2r) [inline commonutils/publish.gradle]: Fixed

**Comment by streetsofboston** [comment]
Ship it.

**Comment by gdaniels** [comment]
:shipit:

**Pull request merged**